src/database-controller/deploy/database-controller.yaml.template

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: database-controller-sts
+spec:
+  selector:
+    matchLabels:
+      app: database-controller
+  serviceName: database-controller
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: database-controller
+    spec:
+      serviceAccountName: database-controller-account
+      hostNetwork: true
+      initContainers:
+      - name: database-initializer
+        image: {{ cluster_cfg["cluster"]["docker-registry"]["prefix"] }}database-controller:{{ cluster_cfg["cluster"]["docker-registry"]["tag"] }}
+        imagePullPolicy: Always
+        env:
+        - name: LOG_LEVEL
+          value: "{{ cluster_cfg['database-controller']['log-level'] }}"
+        - name: K8S_CONNECTION_TIMEOUT_SECOND
+          value: "{{ cluster_cfg['database-controller']['k8s-connection-timeout-second'] }}"
+        - name: WRITE_MERGER_CONNECTION_TIMEOUT_SECOND
+          value: "{{ cluster_cfg['database-controller']['write-merger-connection-timeout-second'] }}"
```

Comment:
Do you also implement the  [ApiServerRetain Mode]:
```
Do not delete exceeded Frameworks in ApiServer
DB Service [ApiServerRetain Mode] ->
As the exceeded Frameworks are not visible to user, but still occupy resources, in [Default: Normal Mode], we should auto delete them.
```

Reply:
And do you have a test case for the [Normal Mode], i.e. delete jobs which in apiserver but not in DB

---
In reply to: [466121708](https://github.com/microsoft/pai/pull/4761#discussion_r466121708) [](ancestors = 466121708)

Reply:
Done.